editor/src/components/editor/actions/actions.tsx

```diff
@@ -5607,3 +5623,96 @@ export function insertWithReparentStrategies(
     newPaths: newPaths,
   }
 }
+
+function getRepositionCoordinatesAndGroupTrueUp(
+  jsxMetadata: ElementInstanceMetadataMap,
+  pathTrees: ElementPathTrees,
+  reparentTargetPath: ElementPath,
+  elementToInsert: PasteElementToInsert,
+  element: ElementInstanceMetadata | null,
+): {
+  groupTrueUpPaths: ElementPath[]
+  repositionCoordinates: CanvasPoint
+} {
+  const reparentTargetParentIsGroup = treatElementAsGroupLike(
+    jsxMetadata,
+    pathTrees,
+    reparentTargetPath,
+  )
+  const maybeElementAncestorGroup =
+    EP.getAncestors(elementToInsert.elementPath).find((path) => {
+      return treatElementAsGroupLike(jsxMetadata, pathTrees, path)
+    }) ?? null
+
+  function getCoordinates(): CanvasPoint {
+    if (!reparentTargetParentIsGroup && maybeElementAncestorGroup == null) {
+      // no groups are involved
+      return elementToInsert.intendedCoordinates
+    }
+
+    const elementToInsertFrame = element?.globalFrame ?? null
+    if (elementToInsertFrame == null || !isFiniteRectangle(elementToInsertFrame)) {
+      // this should never happen
+      return elementToInsert.intendedCoordinates
+    }
+
+    if (reparentTargetParentIsGroup) {
+      const groupFrame =
+        MetadataUtils.findElementByElementPath(jsxMetadata, reparentTargetPath)?.globalFrame ?? null
```

Comment:
instead of the globalFrame, we should be using `specialSizeMeasurements.globalContentBoxForChildren`